nodepool/config/templates/stack-template.json

```diff
@@ -161,7 +161,7 @@
         },
         "ImageId": "{{.AMI}}",
         "InstanceType": "{{.WorkerInstanceType}}",
-        "KeyName": "{{.KeyName}}",
+        {{if .KeyName}}"KeyName": "{{.KeyName}}",{{end}}
```

Comment:
Could you add a test to cover the case in which `KeyName` is missing to our integration test suite for node pools: `aws_test.go`?
The suite runs validations on cfn stack templates calling actual AWS APIs.
That way we can at least save ourselves from breaking things badly in the future.